roles/translator/src/downstream_sv1/downstream.rs

```diff
@@ -79,80 +85,155 @@ impl Downstream {
         }));
         let self_ = downstream.clone();
 
+        let (tx_shutdown, rx_shutdown): (Sender<bool>, Receiver<bool>) = async_channel::bounded(3);
+
         // Task to read from SV1 Mining Device Client socket via `socket_reader`. Depending on the
         // SV1 message received, a message response is sent directly back to the SV1 Downstream
         // role, or the message is sent upwards to the Bridge for translation into a SV2 message
         // and then sent to the SV2 Upstream role.
-        task::spawn(async move {
+        let rx_shutdown_clone = rx_shutdown.clone();
+        let tx_shutdown_clone = tx_shutdown.clone();
+        let _socket_reader_task = task::spawn(async move {
             loop {
+                task::sleep(std::time::Duration::from_millis(5)).await;
```

Comment:
Can you comment on why 1.) there is a sleep here and 2.) why it's 5ms - it would be good to capture this context in code for our future selves

Reply:
Should I link this issue in the code as well? https://github.com/stratum-mining/stratum/issues/399